src/utils/DeviceUtils.ts

```diff
@@ -0,0 +1,24 @@
+import { Platform } from 'react-native';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import DeviceInfo from 'react-native-device-info';
+
+const DEVICE_ID_KEY = 'device_id';
+
+export async function getDeviceId(): Promise<string> {
+    try {
+        // Try to get cached device ID
+        let deviceId = await AsyncStorage.getItem(DEVICE_ID_KEY);
+        
+        if (!deviceId) {
+            // If no cached ID, generate a new one using device info
+            deviceId = `${Platform.OS}-${await DeviceInfo.getUniqueId()}`;
+            await AsyncStorage.setItem(DEVICE_ID_KEY, deviceId);
+        }
+        
+        return deviceId;
+    } catch (error) {
+        console.error('Error getting device ID:', error);
+        // Fallback to a timestamp-based ID if everything fails
+        return `${Platform.OS}-${Date.now()}`;
```

Comment:
Are we expecting an issue with `AsyncStorage.setItem`?
The operation should be atomic - save it to storage and return that value OR do not store and return null

Reply:
not with AsyncStorage.setItem but with the deviceId = `${Platform.OS}-${await DeviceInfo.getUniqueId()}`;
When fetching uniqueId fails, use an alternative value but do not store

Reply:
Still, there is an issue here. Consider the case where it goes to `catch` block - in this case 
- the method will return `${Platform.OS}-${Date.now()}` but will not store it in storage
- but `AsyncStorage.getItem(DEVICE_ID_KEY)` will not return anything (catch black should also store the fallback ID in storage)

Have thought on this, if it makes sense update it.